raiden/transfer/mediated_transfer/mediator.py

```diff
@@ -1329,6 +1232,11 @@ def handle_offchain_secretreveal(
     )
     is_secret_unknown = mediator_state.secret is None
 
+    if not mediator_state.transfers_pair:
+        # This will not happen during normal operation, but attackers might
+        # send weird messages.
+        return TransitionResult(mediator_state, list())
+
```

Comment:
Is this meaning if malformed state arrives, do nothing? 

Can'T this potentially happen to all states expecting a message? Is this handled for other messages as well?

Reply:
I added this to address one test which simulates an attacker. Without this check, the attacker could cause a crash. This seemed to be the most simple approach to handle that case.

I'm not sure about the more general questions, but I don't expect this to introduce a problem we didn't have before.